kits/java/src/main/java/com/luxai/lux/Factory.java

```diff
@@ -0,0 +1,47 @@
+package com.luxai.lux;
+
+public class Factory {
+
+    public static final int BUILD_LIGHT = 0;
+    public static final int BUILD_HEAVY = 1;
+    public static final int WATER = 2;
+
+    public static final int X = 0;
+    public static final int Y = 1;
+
+    public int[] pos;
+    public int power;
+    public int strain_id;
+    public int team_id;
+    public String unit_id;
+    public Cargo cargo;
+
+    public boolean canBuildHeavy(Environment environment) {
+        RobotInfo robot = environment.ROBOTS.get("HEAVY");
+        return this.power >= robot.POWER_COST && this.cargo.metal >= robot.METAL_COST;
+    }
+
+    public boolean canBuildLight(Environment environment) {
+        RobotInfo robot = environment.ROBOTS.get("LIGHT");
+        return this.power >= robot.POWER_COST && this.cargo.metal >= robot.METAL_COST;
+    }
+
+    public int waterCost(Obs obs, Environment environment) {
+        int lichenCounter = 0;
+        for (int[] row : obs.board.lichen_strains) {
+            for (int lichenStrain : row) {
+                if (lichenStrain == this.strain_id)
+                    lichenCounter++;
+            }
+        }
+        return (int) (Math.ceil(lichenCounter / 10.0) * environment.LICHEN_WATERING_COST_FACTOR);

```

Comment:
This seems incorrect. It should be `Math.ceil(lichenCounter / environment.LICHEN_WATERING_COST_FACTOR)`.

Reply:
a few comments above StoneT2000 confirmed that it is correct

Reply:
@StoneT2000 Maybe I'm missing something but the python kit is still different https://github.com/Lux-AI-Challenge/Lux-Design-2022/blob/ea8a918d06ec16baaa8423d0f622ace7f7708d18/kits/python/lux/factory.py#L52